client/src/components/InputElements/FlowsFilterInput.tsx

```diff
@@ -0,0 +1,144 @@
+import { Button, Classes, Icon, MenuItem } from '@blueprintjs/core';
+import { ItemRenderer, MultiSelect } from '@blueprintjs/select';
+import { trim } from 'lodash';
+import React, { useCallback, useState } from 'react';
+import {
+  FlowsFilterDirection,
+  FlowsFilterEntry,
+  FlowsFilterUtils,
+} from '~/domain/flows';
+import css from './FlowsFilterInput.scss';
+
+interface Props {
+  filters: FlowsFilterEntry[];
+  onChange: (filters: FlowsFilterEntry[]) => void;
+}
+
+const FilterMultiSelect = MultiSelect.ofType<FlowsFilterEntry>();
+
+export const FlowsFilterInput = (props: Props) => {
+  const [userInput, setUserInput] = useState<string>('');
+
+  const onClear = useCallback(() => props.onChange([]), []);
+
+  const renderCreateNewItem = useCallback(
+    (
+      query: string,
+      active: boolean,
+      handleClick: React.MouseEventHandler<HTMLElement>,
+    ) => {
+      if (props.filters.some(({ query: text }) => text === query)) {
+        return undefined;
+      }
+
+      return (
+        <MenuItem
+          onClick={handleClick}
+          active={active}
+          text={`Search "${query}"`}
+          icon="search"
+        />
+      );
+    },
+    [props.filters],
+  );
+
+  const handleQueryChange = useCallback((query: string) => {
+    setUserInput(query.replace(/\s/, ''));
+  }, []);
+
+  const handleItemSelect = useCallback(
+    (item: FlowsFilterEntry) => {
+      if (trim(item.query).length === 0) {
+        return;
+      }
+      props.onChange([...props.filters, item]);
+      setUserInput('');
+    },
+    [props.filters, userInput],
+  );
+
+  const handleTagDelete = useCallback(
```

Comment:
Btw it looks quiet weird, better using `// prettier-ignore` with following formatting:
```ts
useCallback(() => {
  // fn's code
}, [dep1, dep2, dep3]);
```